File: herencia-dto/HerenciaDto-Java-IvanRC/src/main/java/models/Persona.java
package models;

public abstract class Persona {
    static String nombre;

    public Persona(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public static <T> int getNombreLength(T t) {
        return nombre.length();
    }
}

File: herencia-dto/SergioDeIscar-herencia-kotlin/SergioDeIscar-herencia-kotlin/src/main/kotlin/models/Persona.kt
package models

import java.io.Serializable

abstract class Persona(
    var id: Long = 0, // Lo he tenido que poner var para que funcione el copy...
    val nombre: String
): Serializable {
    abstract fun toCsvRow(): String
    //Copy method like data class
    abstract fun copy(id: Long = this.id, nombre: String = this.nombre): Persona
}
